File: EcommerceApp-Backend/src/main/java/com/EcommerceApp/dto/Purchase.java
package com.EcommerceApp.dto;


import com.EcommerceApp.entity.Address;
import com.EcommerceApp.entity.Customer;
import com.EcommerceApp.entity.Order;
import com.EcommerceApp.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: EcommerceApp-Frontend/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { Orderitem } from "./orderitem";

export class Purchase {
  customer: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItems: Orderitem[];

  constructor(purchase?: Partial<Purchase>) {
    this.customer = purchase?.customer || new Customer();
    this.shippingAddress = purchase?.shippingAddress || new Address();
    this.billingAddress = purchase?.billingAddress || new Address();
    this.order = purchase?.order || new Order();
    this.orderItems = purchase?.orderItems || [];
  }
}
